webapp/src/modules/asset/utils.js

```diff
@@ -0,0 +1,42 @@
+import { eth } from 'decentraland-eth'
+
+import { ASSET_TYPES } from 'shared/asset'
+import { splitCoordinate } from 'shared/coordinates'
+import { getContractAddress } from 'modules/wallet/utils'
+
+export function getNFTAddressByType(type) {
+  if (type === ASSET_TYPES.parcel) {
+    return getContractAddress('LANDRegistry')
+  } else if (type === ASSET_TYPES.estate) {
+    return getContractAddress('EstateRegistry')
+  }
+  throw Error('invalid asset type')
+}
+
+export async function buildAsset(assetId, assetType, estates) {
```

Comment:
I don't like this the last argument. It's conditional to assetType, and it tomorrow a new assetType appears that needs part of the state like estates does, we don't have a place to add it. Also this is forcing the caller to **always** instantiate all estates before building a random asset.
It made sense before to read the current state and build the asset from there.

A solution might be exporting this helper directly from `/modules/asset/sagas`, just using store.getState(), or something else

cc: @cazala